dd-trace throws error, cannot find module 'retry'

**Describe the bug**
I see this error on starting my app, coming from dd-trace:
```
Error: Cannot find module 'retry'
Require stack:
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/exporters/agent.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/config.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/profiler.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/index.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiler.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/proxy.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/index.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/index.js
- /app/apps/pdf-gen/build/logging/tracing.js
- /app/apps/pdf-gen/build/logging/index.js
- /app/apps/pdf-gen/build/app.js
- /app/apps/pdf-gen/build/server.js
```

I notice `https://github.com/DataDog/dd-trace-js/blob/7ffe1fb7dfe01d2ce0d1108406de25ef84557004/packages/dd-trace/src/profiling/exporters/agent.js` is importing "retry":
```
const retry = require('retry')
```

Is there awareness that `retry` is a devDependency? If `exporters/agent.js` is expected to be used at runtime then it looks to me retry should be made a full dependency? Or am I missing something?

**Environment**
Node 16

* **Operation system:** MacOS
* **Node version:** Node 16
* **Tracer version:**: ^1.3.0
* **Agent version:**: 7

--- a/LICENSE-3rdparty.csv
+++ b/LICENSE-3rdparty.csv
@@ -18,6 +18,7 @@ require,multer,MIT,Copyright 2014 Hage Yaapa
 require,opentracing,MIT,Copyright 2016 Resonance Labs Inc
 require,path-to-regexp,MIT,Copyright 2014 Blake Embrey
 require,performance-now,MIT,Copyright 2013 Braveg1rl
+require,retry,MIT,Copyright 2011 Tim Koschützki Felix Geisendörfer
 require,semver,ISC,Copyright Isaac Z. Schlueter and Contributors
 require,source-map,BSD-3-Clause,Copyright 2009-2011 Mozilla Foundation and contributors
 require,source-map-resolve,MIT,Copyright 2014-2020 Simon Lydell 2019 Jinxiang
@@ -51,7 +52,6 @@ dev,msgpack-lite,MIT,Copyright 2015 Yusuke Kawasaki
 dev,nock,MIT,Copyright 2017 Pedro Teixeira and other contributors
 dev,nyc,ISC,Copyright 2015 Contributors
 dev,proxyquire,MIT,Copyright 2013 Thorsten Lorenz
-dev,retry,MIT,Copyright 2011 Tim Koschützki Felix Geisendörfer
 dev,rimraf,ISC,Copyright Isaac Z. Schlueter and Contributors
 dev,sinon,BSD-3-Clause,Copyright 2010-2017 Christian Johansen
 dev,sinon-chai,WTFPL and BSD-2-Clause,Copyright 2004 Sam Hocevar 2012–2017 Domenic Denicola

--- a/package.json
+++ b/package.json
@@ -75,6 +75,7 @@
     "opentracing": ">=0.12.1",
     "path-to-regexp": "^0.1.2",
     "performance-now": "^2.1.0",
+    "retry": "^0.10.1",
     "semver": "^5.5.0",
     "source-map": "^0.7.3",
     "source-map-resolve": "^0.6.0"
@@ -110,7 +111,6 @@
     "nock": "^11.3.3",
     "nyc": "^15.1.0",
     "proxyquire": "^1.8.0",
-    "retry": "^0.10.1",
     "rimraf": "^3.0.0",
     "sinon": "^11.1.2",
     "sinon-chai": "^3.7.0",

